internal/services/datafactory/data_factory_dataset_azure_sql_table_resource.go

```diff
@@ -0,0 +1,436 @@
+					for i, v := range data.Annotations {
+						annotations[i] = v
+					}
+					azureSqlTable.Annotations = &annotations
+				} else {
+					azureSqlTable.Annotations = nil
+				}
+			}
+
+			if metadata.ResourceData.HasChange("additional_properties") {
+				azureSqlTable.AdditionalProperties = data.AdditionalProperties
+			}
+
+			if metadata.ResourceData.HasChange("schema_column") {
+				azureSqlTable.Structure = data.SchemaColumn
+			}
+
+			dataset = datafactory.DatasetResource{
+				Type:       pointer.To(string(datafactory.TypeBasicDatasetTypeAzureSQLTable)),
+				Properties: azureSqlTable,
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.FactoryName, id.Name, dataset, ""); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (DataFactoryDatasetAzureSQLTableResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			d := metadata.ResourceData
+			client := metadata.Client.DataFactory.DatasetClient
+			id, err := parse.DataSetID(d.Id())
+			if err != nil {
+				return err
+			}
+
+			dataFactoryId := factories.NewFactoryID(id.SubscriptionId, id.ResourceGroup, id.FactoryName)
+			var state DataFactoryDatasetAzureSQLTableResourceSchema
+
+			resp, err := client.Get(ctx, id.ResourceGroup, id.FactoryName, id.Name, "")
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state.Name = id.Name
+			state.DataFactoryId = dataFactoryId.ID()
+
+			azureSqlTable, ok := resp.Properties.AsAzureSQLTableDataset()
+			if !ok {
+				return fmt.Errorf("classifying Data Factory Dataset Azure SQL Table %s: Expected: %q Received: %T", *id, datafactory.TypeBasicDatasetTypeAzureSQLTable, resp.Properties)
+			}
+
+			state.AdditionalProperties = azureSqlTable.AdditionalProperties
+
+			if azureSqlTable.Description != nil {
+				state.Description = pointer.From(azureSqlTable.Description)
+			}
+
+			state.Parameters = flattenDataSetParameters(azureSqlTable.Parameters)
+			state.Annotations = flattenDataFactoryAnnotations(azureSqlTable.Annotations)
+
+			if linkedService := azureSqlTable.LinkedServiceName; linkedService != nil && linkedService.ReferenceName != nil {
+				state.LinkedServiceId = parse.NewLinkedServiceID(id.SubscriptionId, id.ResourceGroup, id.FactoryName, *linkedService.ReferenceName).ID()
+			}
+
+			if properties := azureSqlTable.AzureSQLTableDatasetTypeProperties; properties != nil {
+				val, ok := properties.TableName.(string)
+				if !ok {
+					log.Printf("[DEBUG] Skipping `table_name` since it's not a string")
+				} else {
+					state.TableName = val
+				}
+			}
+
+			if folder := azureSqlTable.Folder; folder != nil && folder.Name != nil {
+				state.Folder = pointer.From(folder.Name)
+			}
```

Comment:
Similarly this should become
```suggestion
			folder := ""
			if folder := azureSqlTable.Folder; folder != nil && folder.Name != nil {
				folder = pointer.From(folder.Name)
			}
			state.Folder = folder
```